larq_zoo/literature/real_to_bin_nets.py

```diff
@@ -0,0 +1,494 @@
+                downsample=block % 2 == 0 and block > 3,
+            )
+
+        if self.include_top:
+            x = self._last_block(x, name=f"{self.model_name}_block_10")
+
+        model = tf.keras.Model(
+            inputs=self.image_input, outputs=x, name=self.model_name,
+        )
+
+        # Load weights.
+        if self.weights == "imagenet":
+            model.load_weights(self._get_weights_path())
+        elif self.weights is not None:
+            model.load_weights(self.weights)
+        return model
+
+    def _get_weights_path(self):
+        raise NotImplementedError(f"No stored weights for {self.model_name}")
+
+
+@factory
+class StrongBaselineNetFactory(_SharedBaseFactory):
+    """Constructor for the strong baseline network (Section 4.1 of Martinez et al.)."""
+
+    scaling_r: int = 8
+
+    input_quantizer: QuantizerType = Field(None)
+    kernel_quantizer: QuantizerType = Field(None)
+
+    class LearnedRescaleLayer(tf.keras.layers.Layer):
+        """Implements the learned activation rescaling XNOR-Net++ style.
+
+        This is used to scale the outputs of the binary convolutions in the Strong
+        Baseline networks.
+        [(Bulat & Tzimiropoulos, 2019)](https://arxiv.org/abs/1909.13863)
+        """
+
+        def __init__(self, output_dim: tuple, **kwargs):
+            super().__init__(**kwargs)
+            assert (
+                len(output_dim) == 3
+            ), f"expected 3 dimensional output size, got {len(output_dim)} dimensional instead"
+            self.output_dim = output_dim
+
+        def build(self, input_shapes):
+            self.scale_h = self.add_weight(
+                name="scale_h",
+                shape=(self.output_dim[0], 1, 1),
+                initializer="ones",
+                trainable=True,
+            )
+            self.scale_w = self.add_weight(
+                name="scale_w",
+                shape=(1, self.output_dim[1], 1),
+                initializer="ones",
+                trainable=True,
+            )
+            self.scale_c = self.add_weight(
+                name="scale_c",
+                shape=(1, 1, self.output_dim[2]),
+                initializer="ones",
+                trainable=True,
+            )
+
+            super().build(input_shapes)
+
+        def call(self, inputs, **kwargs):
+            return inputs * (self.scale_h * self.scale_w * self.scale_c)
+
+        def compute_output_shape(self, **kwargs):
+            return self.output_dim
+
+        def get_config(self):
+            return {"output_dim": self.output_dim}
+
+    def _scale_binary_conv_output(
+        self, conv_input: tf.Tensor, conv_output: tf.Tensor, name: str
+    ):
+        """"The way in which the output of the binary convolution is scaled is the only structural difference
+        between the StrongBaseline networks and the Real-to-Binary networks. This function accepts all inputs
+        required for either function."""
```

Comment:
```suggestion
        """Flexible wrapper for the `LearnedRescaleLayer`.
        
        The way in which the output of the binary convolution is scaled is the only
        structural difference between the StrongBaseline networks and the Real-to-Binary
        networks. This function accepts all inputs required for either function.
        """
```

(Edited for max-width of 88 chars.)